### DISCUSSION TITLE
Managing controversial pull requests

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#155] [General governance]
Some pull requests are controversial and lead to an extended discussion. Managing these discussions and leading the pull request to an approval or rejection is something we could formalize a bit.

We could for example say that if a pull request

a) has been ready for review for more than 2 weeks
b) and the contributor took care of all review comments, 

we could bring that specific pull request to the technical steering committee meeting and make a vote on how to proceed.

What do you think?

**Comment by HarelM** [comment]
I've seen the following work well for some features:
1. Start a discussion, provide some overview of the problem statement, some basic solution proposals, and have a discussion about it.
2. Once the discussion is converging to a solution and people are in agreement an issue can be opened or a PR
3. I generally think an issue should be opened to summarize the discussion, and the direction - much like how a PO/PM would open a feature request or a bug - define that from a developer or a MapLibre user perspective.
4. If this is not straight forward, a design should be added, preferably before starting the implementation, as part of the issue.
5. Once the design is approved a PR can be opened with the relevant code changes.

The above shouldn't be new to anyone in the software industry.

This should minimize the frustration of "I wrote the code, and it is not being accepted" I believe.

I understand that this might be an overkill for PRs that are solving a bug by checking for null, or adding some documentation or typescript definitions, and in those cases, this can be skipped I believe, but I would say this should be a must when changing or adding something to the style-spec or something that is changing or breaking a public API - i.e. something that is more than a few lines of code.

The only problem in the above is the "Once the discussion is converging to a solution" and I think the person who would like to push this feature/bug to get in the repo should make sure that happens, and the TSC might be a place where this should be discussed and bring this up in case the discussion doesn't converge.
